noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable/scheduled_value_change.nr

```diff
@@ -68,77 +61,87 @@ impl<T, DELAY> ScheduledValueChange<T, DELAY> {
             // 
             //         historical                         
             //           block                        block horizon      block of change
             //        =====N=================================H=================|=========>
             //              ^                                 ^
             //              |                                 |   
             //              -----------------------------------
-            //                           delay
+            //                        minimum delay
             //
-            //   b) It could be fewer than `delay` blocks away from the historical block number, in which case it would
-            //      become the limiting factor for the time horizon, which would be the block right before the block of
-            //      change (since by definition the value changes at the block of change).
+            //   b) It could be fewer than `minimum_delay` blocks away from the historical block number, in which case 
+            //      the block of change would become the limiting factor for the time horizon, which would equal the 
+            //      block right before the block of change (since by definition the value changes at the block of 
+            //      change).
             //
             //           historical                         block horizon
             //             block     block of change       if not scheduled
             //        =======N=============|===================H=================>
             //                ^           ^                     ^
             //                |     actual horizon              |   
             //                -----------------------------------
-            //                             delay
+            //                        minimum delay
             //      
             // Note that the current implementation does not allow the caller to set the block of change to an arbitrary
             // value, and therefore scenario a) is not currently possible. However implementing #5501 would allow for 
             // this to happen.
 
             // Because historical_block_number < self.block_of_change, then block_of_change > 0 and we can safely
             // subtract 1.
-            min(self.block_of_change - 1, historical_block_number + DELAY)
+            min(
+                self.block_of_change - 1,
+                historical_block_number + minimum_delay
+            )
         }
     }
 
     /// Mutates a scheduled value change by scheduling a change at the current block number. This function is only
     /// meaningful when called in public with the current block number.
-    pub fn schedule_change(&mut self, new_value: T, current_block_number: u32) {
+    pub fn schedule_change(
+        &mut self,
+        new_value: T,
+        current_block_number: u32,
+        delay: u32,
```

Comment:
For consistency with the comments, should this one be the `minimum_delay` as well?